account_statement_import_file/wizard/account_statement_import.py

```diff
@@ -0,0 +1,358 @@
+# Copyright 2004-2020 Odoo S.A.
+# Licence LGPL-3.0 or later (https://www.gnu.org/licenses/lgpl-3.0).
+
+import base64
+import logging
+
+from odoo import _, api, fields, models
+from odoo.exceptions import UserError
+
+from odoo.addons.base.models.res_bank import sanitize_account_number
+
+logger = logging.getLogger(__name__)
+
+
+class AccountStatementImport(models.TransientModel):
+    _name = "account.statement.import"
+    _description = "Import Bank Statement Files"
+
+    statement_file = fields.Binary(
+        required=True,
+        help="Download bank statement files from your bank and upload them here.",
+    )
+    statement_filename = fields.Char()
+
+    def _import_file(self):
+        self.ensure_one()
+        result = {
+            "statement_ids": [],
+            "notifications": [],  # list of text messages
+        }
+        logger.info("Start to import bank statement file %s", self.statement_filename)
+        file_data = base64.b64decode(self.statement_file)
+        self.import_single_file(file_data, result)
+        logger.debug("result=%s", result)
+        if not result["statement_ids"]:
+            raise UserError(
+                _(
+                    "You have already imported this file, or this file "
+                    "only contains already imported transactions."
+                )
+            )
+        self.env["ir.attachment"].create(self._prepare_create_attachment(result))
```

Comment:
@hbrunn's comment about using `attachment_ids` is still missing it seems.

Reply:
@pedrobaeza I will add it.